collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -180,13 +199,26 @@ void configureButtons(FormController formController) {
     }
 
     protected void goUpLevel() {
-        Collect.getInstance().getFormController().stepToOuterScreenEvent();
+        FormController formController = Collect.getInstance().getFormController();
+        boolean shouldShowRepeatGroupPicker = repeatGroupPickerIndex != null;
+
+        if (shouldShowRepeatGroupPicker) {
```

Comment:
How about `If repeatGroupPickerIndex is set it means we're currently displaying a list of repeat instances. If we set repeatGroupPickerIndex to null, we will go back up to the containing repeat group`